Update TOC TAG and Subproject logos

riaankleinhans opened the pull request:
### Pre-submission checklist:

_If you're adding a new project to the repo, please confirm each of these actions has been completed:_

- [X] Artwork is supplied in both PNG and SVG formats.
- [X] SVGs are real vector-based files. Test and optimize them at https://autocrop.cncf.io before submitting.
- [x] Artwork includes 1 layout: icon.
- [X] At least 3 versions of the artwork are provided: color, black, and white.
- [X] Files follow the existing repo folder and naming conventions.
- [X] The correct examples/file.md is updated with a section for the logos added, in A-Z order.
- [X] A link to the new logo section is added to the main README.md, in A-Z order.

_Note_: Updates to this repo require two approvals to be merged in. One from a CNCF staff member, and the other from a project maintainer whose project's artwork is being updated. We can bypass branch protections for PRs made by maintainers.

cjyabraham reviewed (changes requested):
A few comments:
- the [TAG Security](https://github.com/cncf/artwork/blob/update-toc-logos/examples/archived.md#tag-security) images look broken for me
- the TAG App Delivery and TAG Contributor Strategy menu items don't resolve to the correct page anchors
- There are two TAG Storage sections on [the archived page](https://github.com/cncf/artwork/blob/update-toc-logos/examples/archived.md#tag-storage-logos), one looks like it should be labeled "TAG Network"?
- The [new logo tables](https://github.com/cncf/artwork/blob/update-toc-logos/examples/other.md#mentoring-subproject) shouldn't list columns that are empty. Perhaps just leave out the missing ones?
- Could we get SVG versions of the new logos. SVGs are used most often on the web and look the best.

cjyabraham commented:
- Looks like the [Mentoring Subproject](https://github.com/cncf/artwork/blob/update-toc-logos/examples/other.md#mentoring-subproject) white PNG image is broken.
- I think you could remove the "icon" headers from each of the new tables; the logos there are more like "horizontal" or "stacked"
- [TAG Infrastructure](https://github.com/cncf/artwork/blob/update-toc-logos/examples/other.md#tag-infrastructure) has some "TAG Developer Experience" logos in its middle row
- Regarding my previous comment, the menu item links to `#tag-app-delivery-logos` but the anchor should be `#tag-app-delivery`; same with `#tag-contributor-strategy-logos` when it should be `#tag-contributor-strategy`

riaankleinhans commented:
> * Looks like the [Mentoring Subproject](https://github.com/cncf/artwork/blob/update-toc-logos/examples/other.md#mentoring-subproject) white PNG image is broken.

- Updated

> * I think you could remove the "icon" headers from each of the new tables; the logos there are more like "horizontal" or "stacked"

- Good point, updated

> * [TAG Infrastructure](https://github.com/cncf/artwork/blob/update-toc-logos/examples/other.md#tag-infrastructure) has some "TAG Developer Experience" logos in its middle row

- Copy/paste error - updated

> * Regarding my previous comment, the menu item links to `#tag-app-delivery-logos` but the anchor should be `#tag-app-delivery`; same with `#tag-contributor-strategy-logos` when it should be `#tag-contributor-strategy`

- O, now I get it. And all the TAGs was not in the list. Updated the list and fixed all the links

Thank you!

mrbobbytables reviewed (approved):
lgtm, please squash before merge
